lib/iris/_lazy_data.py

```diff
@@ -83,6 +83,17 @@ def _optimum_chunksize(chunks, shape, limit=None, dtype=np.dtype("f4")):
         "chunks = [c[0] for c in normalise_chunks('auto', ...)]".
 
     """
+    if isinstance(chunks, list):
+        chunks = tuple(chunks)
+    if isinstance(shape, list):
+        shape = tuple(shape)
+    return _optimum_chunksize_internals(chunks, shape, limit, dtype)
+
+
+@lru_cache
+def _optimum_chunksize_internals(
```

Comment:
Given that `_optimum_chunksize` as it currently exists states that it only accepts tuples (not other iterables), which are immutable and therefore cacheable, I could instead edit everywhere we call this to enforce that and thus not need the wrapper.

Reply:
This is an interesting one.  There are actually some side effects / external factors going on in this function that are not in the arguments. See line 100 `dask.config.get("array.chunk-size")`. I guess that config value could be changed during runtime? Unlikely, but possible and should invalidate the cache if it does.

Reply:
I've accounted for that externality and couldn't see any others